cdk/lib/newsletters-tool.ts

```diff
@@ -137,6 +139,31 @@ export class NewslettersTool extends GuStack {
 			app: apiAppName,
 		});
 
+		const readOnlyEndpointApiKeyParam = `/${this.stage}/${this.stack}/${apiAppName}/readOnlyEndpointApiKey`;
+		const readOnlyEndpointApiKey = StringParameter.valueForStringParameter(
+			this,
+			readOnlyEndpointApiKeyParam,
+		);
+
+		new ApplicationListenerRule(this, 'ReadOnlyApiHeaderRule', {
+			listener: ec2AppApi.listener,
+			priority: 1,
+			conditions: [
+				ListenerCondition.httpHeader('X-Gu-API-Key', [readOnlyEndpointApiKey]),
```

Comment:
What is an appropriate format for this and might this be a parameter too? ie, we would hide both key and value in the repo